### PR TITLE
Fix operator checks to follow other check_type logic.

### PR DESCRIPTION
**Pull request by pszulczewski**
This PR fixes `operator` check logic and returned results to make them the same as for other check types.
When the check passes then returned data result is empty, only when the check fails data result is returned to indicate why the check failed.
This is the main behavior for other checks and I think operator check should follow the same behavior.

### CODE REVIEW

File: jdiff/check_types.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def _validate(params) -> None:  # type: ignore[override]
         params_key = params.get("params", {}).get("mode")
         params_value = params.get("params", {}).get("operator_data")
 
-        if not params_key or not params_value:
+        if not params_key or params_value is None:
```
Comment (pszulczewski) [inline jdiff/check_types.py:196]: This was making all-same check to fail here when `False` was passed as value.

File: tests/test_operators.py [hunk 2]
```diff
@@ -39,15 +39,22 @@
...
-        True,
+        False,
     ),
 )
 operator_contains = (
     "pre.json",
     "operator",
     {"params": {"mode": "contains", "operator_data": "EVPN"}},
     "result[0].vrfs.default.peerList[*].[$peerAddress$,peerGroup]",
-    ([{"7.7.7.7": {"peerGroup": "EVPN-OVERLAY-SPINE"}}], False),
```
Comment (lvrfrc87) [inline tests/test_operators.py:50]: I do not get this: here we want to get items that contain the word `EVPN`. The result here is what I would expect for `not-contains`

**Review by lvrfrc87 — COMMENTED** [review]
I believe we need to review this along with @chadell. This approach is the negative of what implemented today. I understand that the idea behind that is to unify the way the test return the bool result but I believe it would be confusing adopt this strategy.

**Review by lvrfrc87 — APPROVED** [review]
As discussed, this is now good to go! Thanks for helping me to see this from a different point of view @pszulczewski !

**Comment by lvrfrc87** [comment]
@pszulczewski would you mind to update the CHANGELOG and relevant bits in poetry?

**Review by lvrfrc87 — APPROVED** [review]
LGTM

**Pull request merged**